gravitee-apim-console-webui/src/management/api/portal/user-group-access/groups/api-portal-access-groups.component.spec.ts

```diff
@@ -0,0 +1,200 @@
+import { MatFormFieldHarness } from '@angular/material/form-field/testing';
+import { InteractivityChecker } from '@angular/cdk/a11y';
+import { MatSelectHarness } from '@angular/material/select/testing';
+import { GioSaveBarHarness } from '@gravitee/ui-particles-angular';
+
+import { ApiPortalAccessGroupsComponent } from './api-portal-access-groups.component';
+
+import { CONSTANTS_TESTING, GioHttpTestingModule } from '../../../../../shared/testing';
+import { ApiPortalUserGroupModule } from '../api-portal-user-group.module';
+import { CurrentUserService, UIRouterState, UIRouterStateParams } from '../../../../../ajs-upgraded-providers';
+import { fakeApi } from '../../../../../entities/api/Api.fixture';
+import { Api } from '../../../../../entities/api';
+import { fakeGroup } from '../../../../../entities/group/group.fixture';
+import { Group } from '../../../../../entities/group/group';
+import { User } from '../../../../../entities/user';
+
+describe('ApiPortalAccessGroupsComponent', () => {
+  const API_ID = 'api-id';
+
+  const fakeAjsState = {
+    go: jest.fn(),
+  };
+
+  const currentUser = new User();
+
+  let fixture: ComponentFixture<ApiPortalAccessGroupsComponent>;
+  let loader: HarnessLoader;
+  let httpTestingController: HttpTestingController;
+
+  const init = async (userPermissions: string[]) => {
+    currentUser.userPermissions = userPermissions;
+    await TestBed.configureTestingModule({
+      imports: [NoopAnimationsModule, GioHttpTestingModule, ApiPortalUserGroupModule, MatIconTestingModule],
+      providers: [
+        { provide: UIRouterState, useValue: fakeAjsState },
+        { provide: UIRouterStateParams, useValue: { apiId: API_ID } },
+        { provide: CurrentUserService, useValue: { currentUser } },
+      ],
+    })
+      .overrideProvider(InteractivityChecker, {
+        useValue: {
+          isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+        },
+      })
+      .compileComponents();
+
+    fixture = TestBed.createComponent(ApiPortalAccessGroupsComponent);
+    loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+
+    httpTestingController = TestBed.inject(HttpTestingController);
+
+    fixture.detectChanges();
+  };
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  describe('Groups tab for user with writing rights', () => {
+    beforeEach(async () => await init(['api-definition-u', 'api-gateway_definition-u']));
+
+    it('should show groups', async () => {
+      expectOneGroupList([fakeGroup({ id: 'my_group', name: 'My Group' })]);
+      expectOneApiGet(fakeApi({ id: API_ID }));
+
+      const fillForm = await loader.getHarness(MatFormFieldHarness);
+      fillForm.getLabel().then((label) => expect(label).toEqual('Groups'));
+
+      const selection = await loader.getHarness(MatSelectHarness);
+      await selection.open();
+
+      await selection.clickOptions({ text: 'My Group' });
```

Comment:
You could have written a `ApiPortalAccessGroupsHarness` to wrap UI accessing code into more readable methods. Like replace L90-93 with `harness.selectGroup('My Group')`